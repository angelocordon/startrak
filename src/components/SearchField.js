import React from 'react';
import styled from 'styled-components';
import { rgba } from 'polished';
import { globalVars as vars } from '../styles';
import { Button } from '../components';

const SearchFieldComponentWrapper = styled.div`
  margin: 0 auto;
  max-width: 500px;
`;

const SearchFieldWrapper = styled.div`
  display: flex;
`;

const SearchFieldLabel = styled.label`
  display: block;
  margin-bottom: 0.25rem;
  font-size: 0.8rem;
  font-weight: bolder;
  text-transform: capitalize;
`;

const SearchFieldInput = styled.input`
  display: flex;
  padding-left: 0.5rem;
  width: 100%;
  height: 2rem;
  border: 1px solid ${rgba(vars.black, 0.3)};
  border-radius: 3px 0px 0px 3px;
`;

const SearchButton = styled(Button)`
  display: flex;
  padding: 0 1rem;
  height: 2rem;
  background-color: #2c3e50;
  border: 1px solid ${rgba(vars.black, 0.5)};
  border-radius: 0px 3px 3px 0px;
  color: #fff;
`;

export default function SearchField() {
  return (
    <SearchFieldComponentWrapper>
      <SearchFieldLabel htmlFor="search-field">
        Repository Search Field
      </SearchFieldLabel>
      <SearchFieldWrapper>
        <SearchFieldInput
          type="text"
          name="search-field"
          placeholder="ex: Ruby on Rails"
        />
        <SearchButton>Search</SearchButton>
      </SearchFieldWrapper>
    </SearchFieldComponentWrapper>
  );
}
