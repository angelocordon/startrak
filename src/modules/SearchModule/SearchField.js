import React, { useRef } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { rgba } from 'polished';
import { globalVars as vars } from '../../styles';
import { Button } from '../../components';

const SearchFieldComponentWrapper = styled.div``;

const SearchFieldWrapper = styled.div`
  display: flex;
`;

const SearchFieldLabel = styled.label`
  display: block;
  margin: 0.67em 0;
  font-size: 1rem;
  font-weight: bolder;
  text-transform: capitalize;
`;

const SearchFieldInput = styled.input`
  display: flex;
  padding-left: 0.5rem;
  width: 100%;
  height: 2rem;
  border: 1px solid ${rgba(vars.black, 0.3)};
  border-radius: 3px 0px 0px 3px;
`;

const SearchButton = styled(Button)`
  display: flex;
  padding: 0 1rem;
  height: 2rem;
  background-color: #2c3e50;
  border: 1px solid ${rgba(vars.black, 0.5)};
  border-radius: 0px 3px 3px 0px;
  color: #fff;
`;

export default function SearchField({ onSearch }) {
  const inputRef = useRef();

  const handleSearch = function() {
    onSearch(inputRef.current.value);
  };

  const handleInputKeyPress = function(event) {
    if (event.key === 'Enter') {
      handleSearch();
    }
  };

  return (
    <SearchFieldComponentWrapper>
      <SearchFieldLabel htmlFor="search-field">
        Search Repositories
      </SearchFieldLabel>
      <SearchFieldWrapper>
        <SearchFieldInput
          type="text"
          name="search-field"
          placeholder="ex: Ruby on Rails"
          ref={inputRef}
          onKeyPress={handleInputKeyPress}
        />
        <SearchButton onClick={handleSearch}>Search</SearchButton>
      </SearchFieldWrapper>
    </SearchFieldComponentWrapper>
  );
}

SearchField.propTypes = {
  onSearch: PropTypes.func,
};
