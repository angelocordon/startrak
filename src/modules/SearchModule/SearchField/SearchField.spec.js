import React from 'react';
import { shallow } from 'enzyme';
import SearchField from './index';

describe('<SearchField />', () => {
  const wrapper = shallow(<SearchField />);

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('<SearchField /> elements', () => {
    const Component = wrapper.render();
    const inputField = Component.find('input');
    const searchButton = Component.find('button');

    it('should properly render', () => {
      expect(inputField).toHaveLength(1);
      expect(searchButton).toHaveLength(1);
    });
  });
});
