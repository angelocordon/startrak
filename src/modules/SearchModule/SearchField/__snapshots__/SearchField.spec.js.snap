// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchField /> should render correctly 1`] = `ShallowWrapper {}`;
